src/adagrams.js

```diff
@@ -1,15 +1,170 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+import _, { map } from "underscore";
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+const lettersPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
 };
-
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
+const letterScore = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
 };
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+//Solution 1
+// create class and using static method
+class Adagrams {
+  static drawLetters() {
+    // Implement this method for wave 1
+    // creating list to store all posibilities of each letter
+    const lettersPoolList = [];
+    for (let letter in lettersPool) {
+      for (let index = 0; index < lettersPool[letter]; index++) {
+        lettersPoolList.push(letter);
+      }
+    }
+
+    // drawing 10 random letter into a list
+    let randomLetter = _.sample(lettersPoolList, 10);
+    return randomLetter;
+  }
+
+  static usesAvailableLetters(input, lettersInHand) {
+    // Implement this method for wave 2
+    // deep copy letterInHand
+    let copyLetterInHand = lettersInHand.slice();
```

Comment:
Remember, `const` doesn't make a _value_ immutable it just means that you can't _reassign_ that variable.

Ideally this would be declared as a `const`:

```suggestion
    const copyLetterInHand = lettersInHand.slice();
```